modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/agg/MapReduceAggregates.java

```diff
@@ -356,10 +486,190 @@ private static MapReduceAgg createSimpleAgg(AggregateCall call, int reduceArgume
     }
+
+    private static MapReduceAgg createAvgAgg(
+            RelOptCluster cluster,
+            AggregateCall call,
+            int reduceArgumentOffset,
+            RelDataType inputType,
+            boolean canBeNull
+    ) {
+        RelDataTypeFactory tf = cluster.getTypeFactory();
+        RelDataTypeSystem typeSystem = tf.getTypeSystem();
+
+        RelDataType fieldType = inputType.getFieldList().get(call.getArgList().get(0)).getType();
+
+        // In case of AVG(NULL) return a simple version of an aggregate, because result is always NULL.
+        if (fieldType.getSqlTypeName() == SqlTypeName.NULL) {
+            return createSimpleAgg(call, reduceArgumentOffset);
+        }
+
+        // AVG(x) : SUM(x)/COUNT0(x)
+        // MAP    : SUM(x) / COUNT(x)
+
+        // SUM(x) as s
+        RelDataType mapSumType = typeSystem.deriveSumType(tf, fieldType);
+        if (canBeNull) {
+            mapSumType = tf.createTypeWithNullability(mapSumType, true);
+        }
+
+        AggregateCall mapSum0 = AggregateCall.create(
+                SqlStdOperatorTable.SUM,
+                call.isDistinct(),
+                call.isApproximate(),
+                call.ignoreNulls(),
+                ImmutableList.of(),
+                call.getArgList(),
+                call.filterArg,
+                null,
+                call.collation,
+                mapSumType,
+                "AVG" + reduceArgumentOffset + "_MAP_SUM");
+
+        // COUNT(x) as c
+        RelDataType mapCountType = tf.createSqlType(SqlTypeName.BIGINT);
+
+        AggregateCall mapCount0 = AggregateCall.create(
+                SqlStdOperatorTable.COUNT,
+                call.isDistinct(),
+                call.isApproximate(),
+                call.ignoreNulls(),
+                ImmutableList.of(),
+                call.getArgList(),
+                call.filterArg,
+                null,
+                call.collation,
+                mapCountType,
+                "AVG" + reduceArgumentOffset + "_MAP_COUNT");
+
+        // REDUCE : SUM(s) as reduce_sum, SUM0(c) as reduce_count
+        List<Integer> reduceSumArgs = List.of(reduceArgumentOffset);
+
+        // SUM0(s)
+        RelDataType reduceSumType = typeSystem.deriveSumType(tf, mapSumType);
+        if (canBeNull) {
+            reduceSumType = tf.createTypeWithNullability(reduceSumType, true);
+        }
+
+        AggregateCall reduceSum0 = AggregateCall.create(
+                SqlStdOperatorTable.SUM,
+                call.isDistinct(),
+                call.isApproximate(),
+                call.ignoreNulls(),
+                ImmutableList.of(),
+                reduceSumArgs,
+                // there is no filtering on REDUCE phase
+                -1,
+                null,
+                call.collation,
+                reduceSumType,
+                "AVG" + reduceArgumentOffset + "_RED_VAL_SUM");
```

Comment:
such a name pattern seems too verbose. Let's have a look at output of EXPLAIN:

```
    ReduceHashAggregate(group=[{0, 1}], SUM_QTY=[SUM($2)], SUM_BASE_PRICE=[SUM($3)], SUM_DISC_PRICE=[SUM($4)], SUM_CHARGE=[SUM($5)], AVG6_RED_VAL_SUM=[SUM($6)], AVG6_RED_COUNT_SUM=[$SUM0($7)], AVG8_RED_VAL_SUM=[SUM($8)], AVG8_RED_COUNT_SUM=[$SUM0($9)], AVG10_RED_VAL_SUM=[SUM($10)], AVG10_RED_COUNT_SUM=[$SUM0($11)], COUNT_12_MAP_SUM=[$SUM0($12)]), id = 211
```
* AVG -- this is an aggregate node, so everything is agg call. 
* RED -- again, I can derive this from node itself
* VAL -- have no idea why do we need this
* SUM -- seems like this relates to the aggregating function

Do we really need all this parts, or we may short this to something like `"_SUM" + reduceArgumentOffset`? WDYT?

Reply:
Sure, I was thinking about removing this as well.